engine/apps/alerts/models/channel_filter.py

```diff
@@ -126,29 +137,36 @@ def select_filter(cls, alert_receive_channel, raw_request_data, force_route_id=N
                 )
                 pass
 
-        filters = cls.objects.filter(alert_receive_channel=alert_receive_channel)
+        channel_filters = cls.objects.filter(alert_receive_channel=alert_receive_channel)
 
-        satisfied_filter = None
-        for _filter in filters:
-            if satisfied_filter is None and _filter.is_satisfying(raw_request_data):
-                satisfied_filter = _filter
+        for channel_filter in channel_filters:
+            if channel_filter.is_satisfying(raw_request_data, alert_labels):
+                return channel_filter
+        return None
 
-        return satisfied_filter
+    def is_satisfying(
+        self, raw_request_data: "Alert.RawRequestData", alert_labels: typing.Optional["Labels"] = None
+    ) -> bool:
+        return self.is_default or self.check_filter(raw_request_data, alert_labels)
 
-    def is_satisfying(self, raw_request_data):
-        return self.is_default or self.check_filter(raw_request_data)
-
-    def check_filter(self, value):
+    def check_filter(
+        self, raw_request_data: "Alert.RawRequestData", alert_labels: typing.Optional["Labels"] = None
+    ) -> bool:
         if self.filtering_term_type == ChannelFilter.FILTERING_TERM_TYPE_JINJA2:
             try:
-                is_matching = apply_jinja_template(self.filtering_term, payload=value)
-                return is_matching.strip().lower() in ["1", "true", "ok"]
+                return templated_value_is_truthy(
```

Comment:
I decided to move the duplicated logic of `<string>.strip().lower() in ["1", "true", "ok"]` (we also duplicated this 3x in the `Alert` model) to [`common.jinja_templater.apply_jinja_template.templated_value_is_truthy`](https://github.com/grafana/oncall/pull/3778/files#diff-e14099bf4a01d2763b4c119bb91a0ab0fba257487ec1157b7aac5867afdf57d0R61-R64)